### PR TITLE
elastic search implemented

### PR DESCRIPTION
**Pull request by codeinpython-org**

### CODE REVIEW

**Review by 50-Course — COMMENTED** [review]
Just went through this, so far looked great. I need to get on my Personal PC to test the app and approve an MR.

Right now, am on my work pc

File: main-backend/task/models.py [hunk 1]
```diff
@@ -12,16 +12,35 @@
...
+    ('AR', 'Article'),
+]
+
 class Post(models.Model):
     title = models.CharField(max_length=100, blank=False, null=False)
-    description = models.TextField(max_length=1000, blank=False, null=False)
+    description = models.TextField(max_length=1000, blank=True, null=False)
     created_by = models.ForeignKey(User, on_delete=models.CASCADE)
     created_on = models.DateTimeField(default=now)
     updated_on = models.DateTimeField(blank=True, null=True)
```
Comment (thechip911) [inline main-backend/task/models.py:38]: use auto_now=True

File: main-backend/task/models.py [hunk 2]
```diff
@@ -12,16 +12,35 @@
...
     slug = models.SlugField(blank=True, null=True, max_length = 100)
     is_edited = models.BooleanField(default=False)
-    post_status	= models.CharField(max_length=10, blank=False, null=True, choices=POST_STATUS, default='is_open')
```
Comment (thechip911) [inline main-backend/task/models.py:24]: blank is False by defult remove redundent code

**Review by thechip911 — COMMENTED** [review]
Minor Comments

File: main-backend/task/management/command/populate_db.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        programming = Category.objects.create(name='Programming')
+
+        # Create articles
+        website_article = Post.objects.create(
+           title='How to code and deploy a website?',
+           created_by=satyajit,
+           type='TU',
+           content='There are numerous ways of how you can deploy a website...',
+        )
+        website_article.save()
```
[the remark this answers is not in the document]
Comment (codeinpython-org) [reply main-backend/task/management/command/populate_db.py:45]: this is for the dummy data test. @thechip911

File: main-backend/task/management/command/populate_db.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        website_article.categories.add(programming, system_administration, seo_optimization)
+
+        google_article = Post.objects.create(
+           title='How to improve your Google rating?',
+           created_by=jess,
+           type='TU',
+           content='Firstly, add the correct SEO tags...',
+        )
+        google_article.save()
```
[the remark this answers is not in the document]
Comment (codeinpython-org) [reply main-backend/task/management/command/populate_db.py:54]: this is for the dummy data test. @thechip911

File: main-backend/task/management/command/populate_db.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        google_article.categories.add(seo_optimization)
+
+        programming_article = Post.objects.create(
+           title='Which programming language is the best?',
+           created_by=jess,
+           type='RS',
+           content='The best programming languages are:\n1) Python\n2) Java\n3) C/C++...',
+        )
+        programming_article.save()
```
[the remark this answers is not in the document]
Comment (codeinpython-org) [reply main-backend/task/management/command/populate_db.py:63]: this is for the dummy data test. @thechip911

File: main-backend/task/management/command/populate_db.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        ubuntu_article = Post.objects.create(
+           title='Installing the latest version of Ubuntu',
+           created_by=johnny,
+           type='TU',
+           content="In this tutorial, we'll take a look at how to setup the latest version of Ubuntu. Ubuntu "
+                   "(/ʊˈbʊntuː/ is a Linux distribution based on Debian and composed mostly of free and open-source"
+                   " software. Ubuntu is officially released in three editions: Desktop, Server, and Core for "
+                   "Internet of things devices and robots.",
+        )
+        ubuntu_article.save()
```
[the remark this answers is not in the document]
Comment (codeinpython-org) [reply main-backend/task/management/command/populate_db.py:75]: this is for the dummy data test. @thechip911

File: main-backend/task/management/command/populate_db.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+        django_article = Post.objects.create(
+           title='Django REST Framework and Elasticsearch',
+           created_by=johnny,
+           type='TU',
+           content="In this tutorial, we'll look at how to integrate Django REST Framework with Elasticsearch. "
+           "We'll use Django to model our data and DRF to serialize and serve it. Finally, we'll index the data "
+           "with Elasticsearch and make it searchable.",
+        )
+        django_article.save()
```
[the remark this answers is not in the document]
Comment (codeinpython-org) [reply main-backend/task/management/command/populate_db.py:86]: this is for the dummy data test. @thechip911

**Pull request merged**